crates/sui-types/src/sui_system_state/epoch_start_sui_system_state.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::{BTreeMap, HashMap};
+
+use crate::base_types::{AuthorityName, EpochId, SuiAddress};
+use crate::committee::{Committee, StakeUnit};
+use anemo::PeerId;
+use multiaddr::Multiaddr;
+use narwhal_config::{Committee as NarwhalCommittee, WorkerCache, WorkerIndex};
+use serde::{Deserialize, Serialize};
+use sui_protocol_config::ProtocolVersion;
+
+/// This type captures the minimum amount of information from SuiSystemState needed by a validator
+/// to run the protocol. This allows us to decouple from the actual SuiSystemState type, and hence
+/// do not need to evolve it when we upgrade the SuiSystemState type.
+/// Evolving EpochStartSystemState is also a lot easier in that we could add optional fields
```

Comment:
> Evolving EpochStartSystemState is also a lot easier in that we could add optional fields

I am not entirely sure we can just add optional fields here, since we use bcs to encode values to DB, we can not evolve the type this way...

Reply:
Ah you are right.
Do you think it's worth also making this type an enum type?

Reply:
Yes, probably make sense to save some trouble in the future

Reply:
Let me merge this and look into versioning it.